lib/media/track/localmediatrack.js

```diff
@@ -129,21 +188,28 @@ function mixinLocalMediaTrack(AudioOrVideoTrack) {
       return this.enable(false);
     }
 
+    restart(constraints) {
```

Comment:
do you think its valid usage to call `localTrack.restart()` I think it would be useful when say a usb device is attached/detached. To support such case we should add line:
``` 
constraints = constraints === `undefined` ? true;
```
if you think that usage is something that we shouldn't allow then we should validate that constrains are not `undefined` - 

Also what happens if constraints given were `{}` ?

Reply:
@makarandp0 You bring up a good point. We can use the default device by setting the constraints as `{}`. I will document this.

I think it makes sense to have `LocalMediaTrack.restart()` re-use existing constraints unless a constraints object is specified.